graylog2-server/src/main/java/org/graylog2/alerts/AlertServiceImpl.java

```diff
@@ -150,45 +141,33 @@ public AlertCondition fromPersisted(Map<String, Object> fields, Stream stream) t
             (String) fields.get("title"));
     }
 
-    private AbstractAlertCondition createAlertCondition(AbstractAlertCondition.Type type,
-                                                        Stream stream,
-                                                        String id,
-                                                        DateTime createdAt,
-                                                        String creatorId,
-                                                        Map<String, Object> parameters,
-                                                        String title) throws AbstractAlertCondition.NoSuchAlertConditionTypeException {
-        switch (type) {
-            case MESSAGE_COUNT:
-                return messageCountAlertFactory.createAlertCondition(stream, id, createdAt, creatorId, parameters, title);
-            case FIELD_VALUE:
-                return fieldValueAlertFactory.createAlertCondition(stream, id, createdAt, creatorId, parameters, title);
-            case FIELD_CONTENT_VALUE:
-                return fieldContentValueAlertFactory.createAlertCondition(stream, id, createdAt, creatorId, parameters, title);
-            default:
-                throw new AbstractAlertCondition.NoSuchAlertConditionTypeException("Unhandled alert condition type: " + type);
-        }
+    private AlertCondition createAlertCondition(String type,
+                                                Stream stream,
+                                                String id,
+                                                DateTime createdAt,
+                                                String creatorId,
+                                                Map<String, Object> parameters,
+                                                String title) throws AbstractAlertCondition.NoSuchAlertConditionTypeException {
+
+        final AlertCondition.Factory factory = this.alertConditionMap.get(type);
+        checkArgument(factory != null, "Unknown alert condition type: " + type);
+
+        return factory.create(stream, id, createdAt, creatorId, parameters, title);
     }
 
     @Override
-    public AbstractAlertCondition fromRequest(CreateConditionRequest ccr, Stream stream, String userId) throws AbstractAlertCondition.NoSuchAlertConditionTypeException {
+    public AlertCondition fromRequest(CreateConditionRequest ccr, Stream stream, String userId) throws AbstractAlertCondition.NoSuchAlertConditionTypeException {
         final String type = ccr.type();
         if (type == null) {
             throw new AbstractAlertCondition.NoSuchAlertConditionTypeException("Missing alert condition type");
```

Comment:
This is the last usage of the `NoSuchAlertConditionTypeException`. Other places now use `IllegalArgumentException` via the `checkArgument()` in `createAlertCondition()`. I think we should either remove the exception class and use `IllegalArgumentException` or use the `NoSuchAlertConditionTypeException` instead of `IllegalArgumentException` in the other places to make it consistent.